### PR TITLE
Update recaptcha index.php

### PR DESCRIPTION
**Pull request by Mickey248**
add recaptcha google api to security
the Picture 1 i add a id and password mistake but it add a nameid in picture2
bot can go that way but recaptcha protect bot

### CODE REVIEW

**Comment by pmeenan** [comment]
Why are you adding recaptcha to the submission form?  If you need it for your own private instance then you can add it but you need a setting to make sure it is disabled by default.

**Comment by Mickey248** [comment]
edit recaptcha again please check again @pmeenan

**Comment by pmeenan** [comment]
You haven't answered WHY you are adding recaptcha to the test submission form.  If it is added because you need it in a private instance then it should be behind a setting in settings.ini.

Something like storing the recaptcha API key in settings.ini and if it is present THEN emit the HTML for adding it to the page.

**Comment by Mickey248** [comment]
I tried to add security by inserting recaptcha API.
It can help protect bot.
sorry i am try to edit a good way i am try edit in recaptcha API in settings.ini
@pmeenan

**Comment by Mickey248** [comment]
i fix  update in settings.ini to try a protect robot @pmeenan 
it a update for link this https://github.com/WPO-Foundation/webpagetest/pull/1040

File: www/index.php [hunk 1]
```diff
@@ -62,6 +62,10 @@
         $conn['plr'] = 0;
     $connectivity['custom'] = $conn;
 }
+$recaptcha = GetSetting['6LfYXTwUAAAAAPyn1ID4UdsXs-ja3P7qPib37asw'];
```
Comment (pmeenan) [inline www/index.php]: No, you "GetSetting" for the name of the setting (like "recaptcha_api_key").  The value in the ini will be the captcha key and should not be hard-coded.

File: www/index.php [hunk 2]
```diff
@@ -62,6 +62,10 @@
         $conn['plr'] = 0;
     $connectivity['custom'] = $conn;
 }
+$recaptcha = GetSetting['6LfYXTwUAAAAAPyn1ID4UdsXs-ja3P7qPib37asw'];
+if ($recaptcha) {
+    echo <recaptcha html with loaded 6LfYXTwUAAAAAPyn1ID4UdsXs-ja3P7qPib37asw>
```
Comment (pmeenan) [inline www/index.php]: what is this?

File: www/index.php [hunk 3]
```diff
@@ -72,6 +76,7 @@
     <head>
         <title>WebPagetest - Website Performance and Optimization Test</title>
         <?php $gaTemplate = 'Main'; include ('head.inc'); ?>
+        <script src='https://www.google.com/recaptcha/api.js'></script>
```
Comment (pmeenan) [inline www/index.php]: The recaptcha api code should only be loaded if the setting is present.

File: www/index.php [hunk 4]
```diff
@@ -597,6 +602,11 @@
                                         <label for="password" style="width: auto;">Password</label>
                                         <input type="text" name="password" id="password" class="text" style="width: 200px;">
                                     </li>
+                                    <li>
+                                           
+                                          <div class="g-recaptcha" data-callback="makeaction" data-sitekey="6LfYXTwUAAAAAPyn1ID4UdsXs-ja3P7qPib37asw"></div>
```
Comment (pmeenan) [inline www/index.php]: The site key should be loaded from settings, not hard-coded.

File: www/index.php (hunk 1 above)
Comment (Mickey248) [reply www/index.php]: thankyou for suggestion

File: www/index.php (hunk 2 above)
Comment (Mickey248) [reply www/index.php]: 6LfYXTwUAAAAAPyn1ID4UdsXs-ja3P7qPib37asw is a api key sorry very much
but i change for hard-coded

File: www/index.php (hunk 3 above)
Comment (Mickey248) [reply www/index.php]: ok today  i fill in settings.ini

File: www/index.php (hunk 4 above)
Comment (Mickey248) [reply www/index.php]: thankyou for help me i change 6LfYXTwUAAAAAPyn1ID4UdsXs-ja3P7qPib37asw to recaptcha_key
that ok?

**Pull request closed without merging**